src/audio/module_adapter/module/waves.c

```diff
@@ -596,95 +652,77 @@ static int waves_codec_init(struct processing_module *mod)
 	int ret = 0;
 	void *response = NULL;
 
-	comp_dbg(dev, "waves_codec_init() start");
+	comp_info(dev, "waves_codec_init() start");
 
 	waves_codec = module_allocate_memory(mod, sizeof(struct waves_codec_data), 16);
 	if (!waves_codec) {
 		comp_err(dev, "waves_codec_init() failed to allocate %d bytes for waves_codec_data",
 			 sizeof(struct waves_codec_data));
-		ret = -ENOMEM;
+		return -ENOMEM;
 	} else {
-		memset(waves_codec, 0, sizeof(struct waves_codec_data));
 		codec->private = waves_codec;
-
 		ret = waves_effect_allocate(mod);
 		if (ret) {
-			module_free_memory(mod, waves_codec);
-			codec->private = NULL;
-		}
-	}
-
-	if (ret) {
-		comp_err(dev, "waves_codec_init() failed %d", ret);
-		return ret;
-	}
-	waves_codec->setup_cfg.avail = false;
-
-	/* copy the setup config only for the first init */
-	if (codec->state == MODULE_DISABLED && codec->cfg.avail) {
-		struct module_config *setup_cfg = &waves_codec->setup_cfg;
-
-		/* allocate memory for set up config */
-		setup_cfg->data = rballoc(0, SOF_MEM_CAPS_RAM, codec->cfg.size);
-		if (!setup_cfg->data) {
-			comp_err(dev, "waves_codec_init(): failed to alloc setup config");
-			module_free_memory(mod, waves_codec);
-			return -ENOMEM;
-		}
-
-		/* copy the setup config */
-		setup_cfg->size = codec->cfg.size;
-		ret = memcpy_s(setup_cfg->data, setup_cfg->size, codec->cfg.data, setup_cfg->size);
-		if (ret) {
-			comp_err(dev, "waves_codec_init(): failed to copy setup config %d", ret);
-			module_free_memory(mod, waves_codec);
-			return ret;
+			goto err;
 		}
-		setup_cfg->avail = true;
 	}
 
 	ret = MaxxEffect_GetMessageMaxSize(waves_codec->effect, &waves_codec->request_max_bytes,
 					      &waves_codec->response_max_bytes);
 
 	if (ret) {
-		comp_err(dev, "waves_codec_init() MaxxEffect_GetMessageMaxSize returned %d", ret);
-		return -EINVAL;
+		comp_err(dev, "cadence_codec_init() MaxxEffect_GetMessageMaxSize returned %d", ret);
+		ret = -EINVAL;
+		goto err;
 	}
 
 	response = module_allocate_memory(mod, waves_codec->response_max_bytes, 16);
 	if (!response) {
-		comp_err(dev, "waves_codec_init() failed to allocate %d bytes for response",
+		comp_err(dev, "cadence_codec_init() failed to allocate %d bytes for response",
 			 waves_codec->response_max_bytes);
-		return -ENOMEM;
+		ret = -ENOMEM;
+		goto err;
 	}
 	waves_codec->response = response;
 
 	comp_dbg(dev, "waves_codec_init() done");
 	return ret;
+
+err:
```

Comment:
please double check, you may missed free more memory here, there should have multiple err label here:

err1:

err2:

err3:

for example: waves_codec->effect memory?